script/script.go

```diff
@@ -25,18 +25,94 @@ import (
 	gjson "layeh.com/gopher-json"
 )
 
-// InitialPoolSize is the number of lua states created initially per route
-var InitialPoolSize int = 3
+const (
+	defaultPoolSize int = 1000
+	genericStateKey     = "generic"
+)
 
-// MaxPoolSize is the number of lua states stored per route - there may be more parallel
-// requests, but only this number is cached.
-var MaxPoolSize int = 10
+func newState() (*lua.LState, error) {
+	L := lua.NewState(lua.Options{
```

Comment:
I had this tuning in mind as a followup of #1792 
Maybe we should not change the defaults within this PR to have a fair measure of how global pool and max size improve(or not) the baseline.

Reply:
I agree, the current state is just to keep the discussions going on.